reference/shaders-msl-no-opt/comp/glsl.std450.comp

```diff
@@ -199,50 +199,50 @@ inline T spvFaceForward(T n, T i, T nref)
 
 kernel void main0(device SSBO& _19 [[buffer(0)]])
 {
-    _19.res = round(_19.f32.x);
-    _19.res = rint(_19.f32.x);
-    _19.res = trunc(_19.f32.x);
-    _19.res = abs(_19.f32.x);
-    _19.ires = abs(_19.s32.x);
-    _19.res = sign(_19.f32.x);
-    _19.ires = sign(_19.s32.x);
-    _19.res = floor(_19.f32.x);
-    _19.res = ceil(_19.f32.x);
-    _19.res = fract(_19.f32.x);
-    _19.res = radians(_19.f32.x);
-    _19.res = degrees(_19.f32.x);
-    _19.res = sin(_19.f32.x);
-    _19.res = cos(_19.f32.x);
-    _19.res = tan(_19.f32.x);
-    _19.res = asin(_19.f32.x);
-    _19.res = acos(_19.f32.x);
-    _19.res = atan(_19.f32.x);
-    _19.res = sinh(_19.f32.x);
-    _19.res = cosh(_19.f32.x);
-    _19.res = tanh(_19.f32.x);
-    _19.res = asinh(_19.f32.x);
-    _19.res = acosh(_19.f32.x);
-    _19.res = atanh(_19.f32.x);
-    _19.res = atan2(_19.f32.x, _19.f32.y);
-    _19.res = pow(_19.f32.x, _19.f32.y);
-    _19.res = exp(_19.f32.x);
-    _19.res = log(_19.f32.x);
-    _19.res = exp2(_19.f32.x);
-    _19.res = log2(_19.f32.x);
-    _19.res = sqrt(_19.f32.x);
-    _19.res = rsqrt(_19.f32.x);
-    _19.res = abs(_19.f32.x);
-    _19.res = abs(_19.f32.x - _19.f32.y);
-    _19.res = sign(_19.f32.x);
-    _19.res = spvFaceForward(_19.f32.x, _19.f32.y, _19.f32.z);
-    _19.res = spvReflect(_19.f32.x, _19.f32.y);
-    _19.res = spvRefract(_19.f32.x, _19.f32.y, _19.f32.z);
```

Comment:
@HansKristian-Work Hi!
Was looking through this PR and got to wonder if it can adversely affect the performance of such load constructs?
E.g.  a lot of per-component loads to the same vector, perhaps distributed in code. I'm concerned whether the compiler manages to coalesce these as well as before.

Reply:
It might be better since Apple's compiler refuses to consider scalar access chains at all. What could happen is that it ends up loading the full vector and only uses one component. There isn't much we can do about this, we're forced to emit this weird code due to buggy compilers.

Reply:
These patterns should already be dealt with at the SPIR-V level I think if the code is loading from the same data multiple times like this.